### DISCUSSION TITLE
Use both Doc and Blog theme

### DISCUSSION THREAD

**Discussion by iqbal125** [shuding/nextra#2528] [Q&A] [answered]
Hello, 

I have an app where I need both a blog and docs. How do I use both themes in the same project?

**Comment by namnguyenthanhwork** [comment, marked as answer]
You install the docs theme, then add the blog folder and make `/blog` look like this: https://github.com/vercel/turborepo/blob/main/docs/pages/blog.mdx

**Reply by finom** [reply]
The file was removed from the repository. Is there a chance that you can show me how it's made today? I'm new to Nextra, want to replace Docusaurus.

**Reply by namnguyenthanhwork** [reply]
@finom I have created the template using both a blog and docs for this. You can refer at https://github.com/namnguyenthanhwork/nextra-personal-website-template. Please give me a ⭐️ if you liked, thank you.

**Reply by finom** [reply]
@namnguyenthanhwork wow, that's amazing. But I'm integrating Nextra to an existing website and I'm unable to just clone your repository unfortunately. Could you please tell me how it works? Is it only about "blog" and "authors" components? Or there is some specific thing I should consider? I think that would be useful for other people also. Thanks!

**Reply by namnguyenthanhwork** [reply]
@finom you can check old commit and see prev version [like this](https://github.com/vercel/turbo/tree/dbd7d5d37cd4070602247afd1ec7d1d17adc9c1d/docs/pages). Then you check and copy logic in `blog.mdx` file.

**Reply by namnguyenthanhwork** [reply]
@finom full code in `blog.mdx` file. You just create `blog.mdx` and add it inside `pages` folder -> done.

```
---
description: The latest updates and releases from the Turbo team at Vercel.
type: posts
---

import { getPagesUnderRoute } from "nextra/context";
import Link from "next/link";

export function BlogHeader() {
  return (
    <div className="max-w-screen-lg mx-auto pt-4 pb-8 mb-16 border-b border-gray-400 border-opacity-20">
      <h1>
        <span className="font-bold leading-tight lg:text-5xl">Blog</span>
      </h1>
      <p className="text-center text-gray-500 dark:text-gray-400 font-space-grotesk">
        The latest updates and releases from the Turbo team at Vercel.
      </p>
    </div>
  );
}

export function BlogIndex() {
  return getPagesUnderRoute("/blog").map((page) => {

    // If the page has a `href` front matter item, we'll render it as an external link
    const externalLinkProps = page.frontMatter.href
      ? {
          href: page.frontMatter.href,
          target: "_blank",
          rel: "noopener noreferrer",
        }
      : {};

    return (
      <div key={page.route} className="mb-10">
        <Link
          href={page.route}
          style={{ color: "inherit", textDecoration: "none" }} className="block font-semibold mt-8 text-2xl"
          {...externalLinkProps}
          >
            {page.meta?.title || page.frontMatter?.title || page.name}
        </Link>
        <p className="opacity-80" style={{ marginTop: ".5rem" }}>
          {page.frontMatter?.description}{" "}
          <span className="inline-block">
            <Link href={page.route}>{"Read more →"}</Link>
          </span>
        </p>
        {page.frontMatter?.date ? (
          <p className="opacity-50 text-sm">{page.frontMatter.date}</p>
        ) : null}
      </div>
    );

});
}

<BlogHeader />
<BlogIndex />
```

**Reply by finom** [reply]
@namnguyenthanhwork are you saying this approach would work in older versions of Nextra only? I tried to copy blog.tsx file and I get bunch of errors such as that:
<img width="847" alt="image" src="https://github.com/shuding/nextra/assets/1082083/32320ae6-c26d-4f01-88fc-d1b977ba724e">

**Reply by finom** [reply]
@namnguyenthanhwork unfortunately I get this when I add blog.mdx to the pages:
<img width="1293" alt="image" src="https://github.com/shuding/nextra/assets/1082083/4d01b3c1-51b6-4970-adb2-db557af6d9de">

**Reply by namnguyenthanhwork** [reply]
@finom You should check to make sure the structure is correct or not, like this
<img width="655" alt="image" src="https://github.com/shuding/nextra/assets/78526193/9532f802-f41d-47dd-954a-eb64a829798d">

In `pages` folder has `blog` folder and `blog.mdx` file. You can clone docs theme, copy `blog.mdx` file and `blog` folder, add them inside `pages` folder and test it.

<img width="1728" alt="image" src="https://github.com/shuding/nextra/assets/78526193/3a407c88-f020-4b2c-afbe-0b869ae4a9a4">

**Reply by finom** [reply]
@namnguyenthanhwork I don't know what did I do wrong but it works now! Thank you!

**Reply by finom** [reply]
@namnguyenthanhwork do you have an idea why `getPagesUnderRoute("/blog")` might return only only one page? When I reload the page it renders as expected. I also tried `getAllPages` and `getCurrentLevelPages` and they all return only a single entry for the blog. My _meta.json is 
```
{
  "vovk-2-0": {
    "title": "Vovk.ts 2.0 is here!",
    "type": "page"
  },
  "vovk-2-1": {
    "title": "Vovk.ts 2.1",
    "type": "page"
  }
}
```


And the function has only vovk-2-0 for some reason. 

The root _meta.json is simple too:

```
...
"blog": {
    "title": "Blog",
    "type": "page"
  },
...
```

Is there any way to "reload" the list of pages? It looks like nextra is poorly maintained and I don't even know if it makes sense to create an issue.

**Reply by finom** [reply]
Made a dirty fix by outputing `<pre>{JSON.stringify(getPagesUnderRoute("/blog"), null, 2)}</pre> ` in dev mode and pasting its contents to a separate JSON file 🤦

**Reply by namnguyenthanhwork** [reply]
@finom You mean the first time the page loads it only creates a single page? (while you have 2 pages). I tried with nextra-docs, the first time it still loaded the correct number of pages you declared

<img width="613" alt="image" src="https://github.com/shuding/nextra/assets/78526193/0c80c95b-2555-475f-8842-abb0e23e0824">

<img width="1720" alt="image" src="https://github.com/shuding/nextra/assets/78526193/dc58c91a-cbac-4103-ab54-05c2d8c38ae8">

**Reply by finom** [reply]
@namnguyenthanhwork yes, but when I deploy and navigate to blog from another menu item, it shown me only one entry. At the same time on development it's completely fine. I'm sure it's not about your approach, it's something with Nextra of Next.js pages router that I haven't used for long time.

**Reply by huykon** [reply]
hi guys, how to use in nextra version 4.x?
